packages/daisy/src/components/button/button.tsx

```diff
@@ -1,76 +1,66 @@
 import { component$, QwikIntrinsicElements, Slot } from '@builder.io/qwik';
 import { Button as HeadlessButton } from '@qwik-ui/headless';
 import { clsq } from '@qwik-ui/shared';
+import { daisyConfig } from './daisy.config';
 
 export type HTMLButtonProps = QwikIntrinsicElements['button'];
 
 export type DaisyButtonProps = {
+  variant?: DaisyButtonVariants
+  size?: DaisyButtonSizes
+  active?: boolean;
   noAnimation?: boolean;
+  circle?: boolean;
   glass?: boolean;
   loading?: boolean;
-  circle?: boolean;
-  square?: boolean;
-  active?: boolean;
-  disabled?: boolean;
   outline?: boolean;
-  size?: 'xs' | 'sm' | 'md' | 'lg' | 'block' | 'wide';
-} & DaisyButtonModifiers;
-
-export type DaisyButtonModifiers = {
-  primary?: boolean;
-  secondary?: boolean;
-  accent?: boolean;
-  info?: boolean;
-  success?: boolean;
-  warning?: boolean;
-  error?: boolean;
-  ghost?: boolean;
-  link?: boolean;
+  square?: boolean;
 }
 
+export type DaisyButtonVariants = 'primary' | 'secondary' | 'accent' | 'info' | 'success' | 'warning' | 'error' | 'ghost' | 'link' | 'disabled';
+export type DaisyButtonSizes = 'xs' | 'sm' | 'md' | 'lg' | 'block' | 'wide';
 export type ButtonProps = HTMLButtonProps & DaisyButtonProps ;
 
 export const Button = component$(
   (props: ButtonProps) => {
     const {
-      class: cls,
-      primary, secondary, accent, info, success, warning, error, ghost, link, outline, active, disabled, glass, loading, noAnimation,
+      size = 'md',
+      variant = 'primary',
       circle,
+      active,
+      class: cls,
+      disabled,
+      glass,
+      loading,
+      noAnimation,
+      outline,
       square,
-      size = 'md',
-      ...rest } = props;
+      ...rest
+    } = props;
+
+    const {
+      variants,
+      sizes,
+      options
+    } = daisyConfig
 
     return (
       <HeadlessButton
         {...rest}
         class={
           clsq(
```

Comment:
Maybe not related to this PR.

I read that `clsq` is a rewrite of `clsx` in TypeScript.
However, it is hard to guess, what `clsq` does.
Even reading the implementation of `clsq` does not help quickly, since it is optimized code.

Can we come up with a better name here?
Additionally, I would add a JSDoc comment, clarifying what the code  of  `clsq` does.
We can provide some Unit Tests as well.

```suggestion
          applyCssClasses(
```

I am glad to help here, if you want, since my comment is a bit out of scope for this issue.

Reply:
Yes you're right.
1. I have thought  to use `clsq` because `clsx` is "famous" lib in the React and other ecosystems.
2. I updated the code from ES5 to TS but since I have changed `arguments` with `rest` operators, var with `let` and so on there could be (even I don't think) some cases that doesn't work like the original one
3. So I also think we should integrate unit tests but @gioboa tell me that we are writing tests in this phase and we'll add them later.
4. I also think  this utility (or something similar) should be part of the Qwik core.

Anyway, in the meanwhile I can rename `cls`. `applyCssClasses` is not bad but other frameworks use:

* Angular: `[ngClass]="..."`
* Vue: `<div :class="{}"`
* Astro: `<div class:list={}`

What do you think about:
`qClass`
or 
`classList` (like the DOM property)
?